src/main/java/edu/harvard/iq/dataverse/license/validation/LicenseErrorHandler.java

```diff
@@ -0,0 +1,75 @@
+package edu.harvard.iq.dataverse.license.validation;
+
+import com.google.common.collect.Lists;
+import edu.harvard.iq.dataverse.license.dto.LicenseDto;
+import edu.harvard.iq.dataverse.license.dto.LocaleTextDto;
+import edu.harvard.iq.dataverse.util.BundleUtil;
+import edu.harvard.iq.dataverse.util.JsfHelper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.validator.routines.UrlValidator;
+
+import javax.ejb.Stateless;
+import javax.faces.application.FacesMessage;
+import javax.faces.context.FacesContext;
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Class responsible for validating if all information's regarding new license are correct.
+ */
+@Stateless
+public class LicenseErrorHandler implements Serializable {
+
+    private UrlValidator urlValidator = new UrlValidator(Lists.newArrayList("http", "https").toArray(new String[0]));
+
+    // -------------------- LOGIC --------------------
+
+    /**
+     * Checks if new license is valid.
+     *
+     * @param licenseDto
+     * @return list of all errors
+     */
+    public List<FacesMessage> validateNewLicenseErrors(LicenseDto licenseDto) {
+        handleLicenseNameError(licenseDto);
+
+        handleLicenseLocalizedNamesErrors(licenseDto);
+
+        handleUrlErrors(licenseDto);
+
+        return FacesContext.getCurrentInstance().getMessageList();
```

Comment:
Hmm you are in that regard that it returns more than it does, but my thought was that if we want to test what kind of messages are added to failed validation context it is useful for testing. Atm it is not really possible to test it with Arquillian since you need "frontend context' in order to initialize FacesContext.